docs/ruleset_functions/get_energy_required_to_heat_swh_use.md

```diff
@@ -0,0 +1,67 @@
+- get the drain heat recovery efficiency if heat is recovered by drain: `if(swh_use.is_heat_recovered_by_drain): drain_heat_recovery_efficiency = distribution.drain_heat_recovery_efficiency`
+- get the use_units: `use_units = swh_use.use_units`
+- convert the shw_water_used from whatever units it's currently in to volume (assumed to be gallons)
+- if the use_units is POWER_PER_PERSON: `if use_units == "POWER_PER_PERSON":`
+  - set power equal to the swh_use_value * space.number_of_occupants: `power = swh_use_value * space.number_of_occupants`
+  - in this case, swh_use_value is assumed to have units of btu/hr/person, so calculate energy_required by multiplying power (btu/hr) by the sum of the hourly_schedule: `energy_required = power * sum(hourly_schedule)`
+  - return energy_required: `return energy_required`
+- otherwise if use_units is POWER_PER_AREA: `elsif use_units == "POWER_PER_AREA":`
+  - set power equal to the swh_use_value * space.floor_area: `power = swh_use_value * space.floor_area`
+  - in this case, swh_use_value is assumed to have units of btu/hr/ft2, so calculate energy_required by multiplying power (btu/hr) by the sum of the hourly_schedule: `energy_required = power * sum(hourly_schedule)`
+  - return energy_required: `return energy_required`
+- otherwise if use_units is POWER: `elsif use_units == "POWER":`
+  - in this case, swh_use_value is assumed to have units of btu/hr, so calculate energy_required by multiplying swh_use_value by the sum of the hourly_schedule: `energy_required = swh_use_value * sum(hourly_schedule)`
+  - return energy_required: `return energy_required`
+- otherwise if use_units is VOLUME_PER_PERSON: `elsif use_units == "VOLUME_PER_PERSON":`
+  - in this case swh_use_value is assumed to have the units of gallons/hr/person:
+  - set volume_flow_rate equal to swh_use_value * space.number_of_occupants: `volume_flow_rate = swh_use_value * space.number_of_occupants`
+- otherwise if use_units is VOLUME_PER_AREA: `elsif use_units == "VOLUME_PER_AREA":`
+  - in this case swh_use_value is assumed to have the units of gallons/hr/ft2:
+  - set volume_flow_rate equal to swh_use_value * space.area: `volume_flow_rate = swh_use_value * space.area`
+- otherwise if use_units is VOLUME: `elsif use_units == "VOLUME":`
+  - in this case swh_use_value is assumed to have the units gallons/hr
+  - set volume_flow_rate equal to swh_use_value: `volume_flow_rate = swh_use_value`
+
+- set energy_required to 0: `energy_required = 0`
+- iterate through each hourly value of the hourly_schedule: `for index, hourly_value in enumerate(hourly_schedule):`
+  - set volume_this_hour equal to volume_flow_rate * hourly_value: `volume_this_hour = volume_flow_rate * hourly_value`
+  - reduce volume_this hour by the recovery efficiency: `volume_this_hour = volume_this_hour * (1-drain_heat_recovery_efficiency)`
+  - calculate the dT for the hour: `dT = temp_at_fixture - inlet_T_sched[index]`
```

Comment:
The temperature at the fixture is a mixure of hot and cold water. If the gallons of water is hot water only then you would need to use the design_supply_temperature data element which is part of the ServiceWaterHeatingDistributionSystem data group instead. Unless .use is mixed hot and cold water. It is not clear from the description.

Reply:
let's discuss on the call

Reply:
issue posted to schema - @JasonGlazer will review what is needed for Title 24.